src/OrleansRuntime/GrainTypeManager/TypeManager.cs

```diff
@@ -1,33 +1,34 @@
 using System;
 using System.Threading.Tasks;
-using Orleans.Runtime.Providers;
+using Orleans.Streams;
 
 namespace Orleans.Runtime
 {
     internal class TypeManager : SystemTarget, ITypeManager
     {
         private readonly GrainTypeManager grainTypeManager;
+        private readonly ImplicitStreamSubscriberTable implicitStreamSubscriberTable;
 
-        internal TypeManager(SiloAddress myAddr, GrainTypeManager grainTypeManager)
+        internal TypeManager(
+            SiloAddress myAddr,
+            GrainTypeManager grainTypeManager,
+            ImplicitStreamSubscriberTable implicitStreamSubscriberTable)
             : base(Constants.TypeManagerId, myAddr)
         {
+            if (implicitStreamSubscriberTable == null) throw new ArgumentNullException(nameof(implicitStreamSubscriberTable));
```

Comment:
That's true. The NRE is not really needed here. The value won't be null even if it hasn't yet been initialized - DI  is used to construct `ImplicitStreamSubscriberTable` and then `TypeManager` is constructed using it inside `Silo.CreateSystemTargets` like so:
```C#
var implicitStreamSubscriberTable = Services.GetRequiredService<ImplicitStreamSubscriberTable>();
this.RegisterSystemTarget(new TypeManager(this.SiloAddress, this.typeManager, implicitStreamSubscriberTable));
```

Currently, it's initialized in `Silo.DoStart` before the call to `CreateSystemTargets`,
On the client side, it's initialized by calling into the gateway (which hits the `TypeManager` system target).

So I believe this is sound